### PR TITLE
(fix) O3-3274: Fix failing expressions

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR introduces a fix for failing expressions within the forms' engine.

## Related Issue

https://openmrs.atlassian.net/jira/software/c/projects/O3/boards/37?assignee=712020:381d2aac-e89b-44ab-803b-02445824ad47&selectedIssue=O3-3274

### CODE REVIEW

**Comment by arodidev** [comment]
@samuelmale the CI disallows importation of the entire `lodash` module, is there a way around this?

**Comment by samuelmale** [comment]
> @samuelmale the CI disallows importation of the entire `lodash` module, is there a way around this?

In as much as there is probably a way to dynamically import lodash, I think it's overkill and maybe unnecessary. 

Question: Do we really need the entire library? Is that what's happening in AFE @ibacher @denniskigen? 

I've done a quick search in Ampath's config repo, and [seems like they're mainly consuming lodash's `isEmpty(...)` utility function](https://github.com/search?q=repo%3AAMPATH%2Fopenmrs-config-amrs+path%3A%2F%5Econfiguration%5C%2Fampathforms%5C%2F%2F+_.&type=code); 

Should we rather aim for a synthetic version of the lodash object?

```ts
import { isEmpty, filter, find } from 'lodash';

const _ = {
  isEmpty,
  filter,
  find,
};

```

**Comment by arodidev** [comment]
@samuelmale I updated the PR with due changes.

File: src/utils/expression-runner.ts [hunk 1]
```diff
@@ -46,14 +47,20 @@ export function evaluateExpression(
...
       return this.value.obs.find((obs) => obs.concept.uuid == concept);
     },
   });
 
   const visitTypeUuid = context.visit?.visitType.uuid ?? '';
 
+  const _ = {
+    isEmpty,
+    filter,
+    find,
```
Comment (samuelmale) [inline src/utils/expression-runner.ts:61]: We actually may not need these two

**Comment by samuelmale** [comment]
@arodidev does this fix all the errors related to expressions even for the "Adult Visit Return" form?

**Comment by arodidev** [comment]
> @arodidev does this fix all the errors related to expressions even for the "Adult Visit Return" form?

@samuelmale this addresses majority of the failing expressions(related to lodash). There are a few failing while referencing field ids, will investigate and address those separately.

**Review by samuelmale — APPROVED** [review]

**Pull request merged**